### PR TITLE
[Bug fix] Various fix from bug bash

### PR DESCRIPTION
**Pull request by mufeili**
### Examples
- PyTorch GraphSAGE
    - Fix the number of "`" in the README file
    - Update the instructions for multi-gpu training
- PyTorch grace
    - Add sklearn dependency 
- PyTorch TAHIN
    - Add sklearn dependency 
    - Fix a minor bug due to a different PyTorch version
- PyTorch graphsaint
    - Add a note in README about the differences between the PPI used in GraphSAINT and DGL's built-in one
- PyTorch ogbn-products GAT
    - Fix a bug
- PyTorch Han
    - Add instruction in README for sampling-based training
- PyTorch GCMC
    - Specify file encoding in `read_csv`

### Doc
- User guide 6.1 EN/CN
    - To access node labels, users should use `blocks[-1].dstdata` rather than `blocks[-1].srcdata`
- `dgl.sampling.node2vec_random_walk`
    - Change `length` to `walk_length` in the code examples
- User guide 6.7
    - Wording
- Multi-GPU graph classification tutorial
    - Reformat and remove extra arguments
- `dgl.adj_product_graph` and `dgl.adj_sum_graph`
    - Interchange the order of `to_simple` and eweight initialization
- `dgl.node_subgraph` and `dgl.edge_subgraph`
    - Fix printed graph information
- `dgl.sampling.sample_neighbors_biased`
    - Fix doc format
- `dgl.reorder_graph`
    - Fix hyperlink for `metis_partition_assignment`
- `multi_update_all`
    - Add an example for udf cross reducer
- `dgl.slice_batch`
    - Add an example
- PyTorch/MXNet/TensorFlow GATConv
    - Replace `dgl.bipartite` with `dgl.heterograph` in example
- PyTorch GCN2Conv
    - Fix math rendering
- PyTorch TWIRLSUnfoldingAndAttention
    - Rename the module to work around the trouble of alias in doc rendering 

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Review by VoVAllen — APPROVED** [review]

**Review by BarclayII — APPROVED** [review]
Oh my.  Thanks a lot.
